crates/ruff/src/rules/flake8_annotations/helpers.rs

```diff
@@ -6,22 +6,32 @@ use ruff_python_semantic::definition::{Definition, Member, MemberKind};
 
 use crate::checkers::ast::Checker;
 
-pub(super) fn match_function_def(stmt: &Stmt) -> (&str, &Arguments, Option<&Expr>, &Vec<Stmt>) {
+pub(super) fn match_function_def(
+    stmt: &Stmt,
+) -> (&str, &Arguments, Option<&Expr>, &Vec<Stmt>, &Vec<Expr>) {
```

Comment:
#4369 introduced named types for each `StmtKind` variant. That means, we can now define custom unions:

```rust
enum MatchFunctionDef<'a> {
	FunctionDef(&'a ast::StmtFunctionDef),
	AsyncFunctionDef(&'a ast::StmtAsyncFunctionDef)
}

impl<'a> MatchFunctionDef<'a> {
	fn cast(stmt: &'a Stmt) -> Option<Self> {
		match &stmt.node {
			StmtKind::FunctionDef(def) => Some(Self::FunctionDef(def)),
			StmtKind::AsyncFunctionDef(def) => Some(Self::AsyncFunctionDef(def)),
			_ => None
		}
	},

	fn name(&self) -> &str {
		match self {
			Self::FunctionDef(def) => def.name(),
			Self::AsyncFunctionDef(def) => def.name(),
		}
	}

	...
}
```

I agree, it's definitely more code but it allows you to retrieve the underlying node when necessary and is more meaningful than a tuple with 5 fields.